pkg/yqutil/yqutil.go

```diff
@@ -94,3 +99,41 @@ func yamlfmt(content []byte) ([]byte, error) {
 	}
 	return formatter.Format(content)
 }
+
+const yamlfmtLineBreakPlaceholder = "#magic___^_^___line"
```

Comment:
I was wondering how the magic string got replaced back by a newline until I realized that we are relying here on the internals of the `yamlfmt` hotfix.

We do have a test to warn us when upstream modifies their hotfix and breaks our piggyback code.

But I think this needs to be documented. And we need to acknowledge that the `adjust` and `replaceLineBreak…` functions are slightly modified versions of the `yamlfmt` code.

I'm also curious about the change in `replaceLineBreaksWithMagicString` relative to the upstream `replaceLineBreakFeature` function (the removal of the `inLineBreaks` logic). I assume this comment is supposed to explain it:

```go
// hotfix: yq does not support line breaks in the middle of a string.
```

It would help to clarify that this refers to the local change because we pass the YAML through yq before passing it to yamlfmt, and is not referring to the "hotfix" logic in yamlfmt.

See other cases where we modified upstream code:


```console
$ ag adapted
pkg/hostagent/dns/dns.go
1:// This file has been adapted from https://github.com/norouter/norouter/blob/v0.6.4/pkg/agent/dns/dns.go

pkg/reflectutil/reflectutil.go
1:// This file has been adapted from https://github.com/containerd/nerdctl/blob/v1.0.0/pkg/reflectutil/reflectutil.go
```